### ISSUE TITLE
Back navigation and hamburger menu 

### ISSUE THREAD

**Issue by gedw99** [axzilla/templui#27]
Just noticed that when you navigate back the hamburger menu opens on the previous page. 

It’s only occurring if the initial page had the hamburger menu open .

it’s off putting ..

https://share.icloud.com/photos/0f4LSgan5SLtbaKEjG3wQfv-A

https://share.icloud.com/photos/0b8bonpXx_66Qhu1zZRiZpT0g

if you do the same on GitHub web, it does not has this behaviour.

the fix could be that in the menu, it closes the menu before the navigation occurs ?

**Comment by axzilla** [comment]
Thank you for reporting this issue! Yes, I've noticed this behavior with the hamburger menu state persisting during back/forward navigation.

I've investigated several approaches to fix this, but haven't found a clean, scalable solution yet. The root cause seems to be how Alpine.js states interact with the browser's back-forward cache (bfcache). This actually affects not just the hamburger menu, but all Alpine.js states (accordions, datepickers, etc).

I've started a discussion in the Alpine.js community to find the best way to handle this pattern, as it needs a solid solution at the framework level.

I'll update this issue once we have more insights from the Alpine.js team and community. Thank you for your patience and for helping improve the library!
